filesystem/templatecopy.go

```diff
@@ -38,7 +53,8 @@ func TemplateCopy(source, destination string, config interface{}) error {
 	return newProcessor(options).process(source, destination)
 }
 
-func templateCopyProcessFileCallBack(sourceFilePath, destinationFilePath string, config interface{}) error {
+func templateCopyProcessFileCallBack(sourceFilePath, destinationFilePath string, addOnConfig interface{}) error {
+	config := addOnConfig.(map[string]interface{})[ConfigKey]
```

Comment:
Would be good to define a type for this, so that it is explicit. We can use common.GetObjFromInterface to get the object from the interface.

Reply:
done